src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystem.java

```diff
@@ -1957,6 +1961,15 @@ private OutputStream getOutputStream(Entry e) throws IOException {
         }
     }
 
+    // Return true if file name is "." or ".."
+    private static boolean isSelfOrParent(final byte[] nameAsBytes) {
+        if (nameAsBytes[0] == '.') {
+            return (nameAsBytes.length == 1) ||
+                    (nameAsBytes.length == 2 && nameAsBytes[1] == '.');
+        }
+        return false;
+    }
```

Comment:
It might be a bit clearer if ZipPath define isSelfOrParent(), that would avoid needing to expose the internal representation.

Reply:
Done. I've updated the PR to implement this suggestion.